docs/content/guides/dagster-pipes/integrating-kubernetes-with-dagster-pipes.mdx

```diff
@@ -1,6 +1,242 @@
+    <code>open_pipes_session</code> approach
+  </a>{" "}
+  instead.
+</Note>
+
+---
+
+## Prerequisites
+
+To use Dagster Pipes with Kubernetes, you’ll need:
+
+- **In the orchestration environment**, you'll need to install the following packages:
+
+  ```shell
+  pip install dagster dagster-webserver dagster-k8s kubernetes
+  ```
+
+  Refer to the [Dagster installation guide](/getting-started/install) for more info.
+
+- **In Kubernetes:**
+  - **An existing Kubernetes cluster.**
+  - **The following information about your Kubernetes cluster**:
+    - `config_file` - The name of the Kubernetes config file
+    - `context` - The name of the Kubernetes context you want to use
+    - `image` - The image you want to run using Dagster Pipes
+
+---
+
+## Step 1: Create an asset computed in Kubernetes
+
+In this step, you’ll create a Dagster asset that, when materialized, opens a Dagster pipes session and spins up a Kubernetes pod that executes some external code.
+
+### Step 1.1: Define the Dagster asset
+
+1. Create a new file named `dagster_k8s_pipes.py` in your Dagster project.
+
+2. Copy and paste the following imports into the file:
+
+   ```python
+   # dagster_k8s_pipes.py
+
+   from dagster import AssetExecutionContext, Definitions, asset
+   from dagster_k8s import PipesK8sClient
+   ```
+
+3. Next, you’ll define the asset. Copy and paste the following below the imports in `dagster_k8s_pipes.py`:
+
+   ```python
+   # dagster_k8s_pipes.py
+
+   @asset
+   def k8s_pipes_asset(context: AssetExecutionContext, k8s_pipes_client: PipesK8sClient):
+     return k8s_pipes_client.run(
+         context=context,
+         image="pipes-materialize:latest",
+     ).get_materialize_result()
+   ```
+
+   Here’s what we did in this example:
+
+   - Created an asset named `k8s_pipes_asset`
+   - Provided <PyObject object="AssetExecutionContext" /> as the `context` argument to the asset. This object provides access to system APIs such as resources, config, and logging. We’ll come back to this a bit later in this section.
+   - Specified a resource for the asset to use, <PyObject module="dagster_k8s" object="PipesK8sClient" />. We’ll also come back to this in a little bit.
+
+     Using the `context` argument, we've passed in the asset's `context` data as well as the Kubernetes image we want to use. The Kubernetes resource used by the asset exposes a `run` method. This method synchronously spins up a Kubernetes pod, runs the specified image, and returns any asset materialization events.
+
+**Note**: Execution can take several minutes even for trivial scripts due to cluster provisioning times.
+
+<!--
+TODO: Takeout if not applicable
+
+When the job finishes executing, Dagster Pipes will return a <PyObject module="dagster_pipes" object="PipesClientCompletedInvocation" /> object. This object exposes a `get_results` method, which you can use to access the results reported by Kubernetes.  -->
```

Comment:
This was originally pulled from the [dagster-databricks README](https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-databricks/README.md#1-pipesdatabricksclient-resource) that Sean wrote. Is this different for K8s?

Reply:
I believe we wanted to move away from `get_results` but @smackesey or @alangenfeld can confirm

Reply:
this is my current understanding of what to use when:
```
@asset -> get_materialization_result
@multi_asset -> get_results
@asset_check -> get_asset_check_result
```

Reply:
I think you could mention this bit about having different result methods for different types of assets or just omit this section from this k8s focused doc